analyze/analyses/dominators/mod.rs

```diff
@@ -47,20 +53,30 @@ pub fn dominators(
             .collect()
     };
 
-    let unreachable_items = if arguments.is_empty() {
-        utils::get_unreachable_items(&items)
-            .map(|item| item.id())
-            .collect::<Vec<ir::Id>>()
-    } else {
-        Vec::new()
-    };
-
     let tree = DominatorTree {
         tree: items.dominator_tree().clone(),
         items: dominator_items,
-        unreachable_items,
         opts: opts.clone(),
+        unreachable_items_summary: summarize_unreachable_items(items, opts),
     };
 
     Ok(Box::new(tree) as Box<traits::Emit>)
 }
+
+fn summarize_unreachable_items(
```

Comment:
outlined this, now that the unreachable items summary is a struct. seemed nicer that way, given that this isn't the primary function of the `dominators` function